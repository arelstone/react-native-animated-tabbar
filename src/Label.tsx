import React, { FC } from 'react';
import { Text, StyleProp, TextStyle, StyleSheet } from 'react-native';

interface LabelProps {
    textStyle?: StyleProp<TextStyle>;
    label: string;
    color: string;
}

const Label: FC<LabelProps> = ({ color, label, textStyle }) => {
    return <Text style={[{ color }, styles.label, textStyle]}>
        {label}
    </Text>;
};

export default Label;


const styles = StyleSheet.create({
    label: {
        textAlign: 'center',
        fontSize: 10,
        paddingTop: 10,
    },
});
