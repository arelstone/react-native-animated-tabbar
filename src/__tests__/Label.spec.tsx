import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import Label from '../Label';

let wrapper: ShallowWrapper;

describe('<Label />', () => {
    beforeAll(() => {
        wrapper = shallow(<Label
            color="blue"
            label="MyLabel"
        />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should display correct text', () => {
        expect(wrapper.find('Text').prop('children')).toEqual('MyLabel');
    });

    it('should apply correct color', () => {
        expect(wrapper.find('Text').prop('style')).toEqual(
            expect.arrayContaining([
                expect.objectContaining({ color: 'blue' }),
            ]),
        );
    });

    it('should apply textStyle', () => {
        wrapper.setProps({ textStyle: { fontSize: 42 } });

        expect(wrapper.find('Text').prop('style')).toEqual(
            expect.arrayContaining([
                expect.objectContaining({ fontSize: 42 }),
            ]),
        );
    });
});
