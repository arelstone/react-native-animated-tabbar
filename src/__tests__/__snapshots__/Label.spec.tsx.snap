// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should match snapshot 1`] = `ShallowWrapper {}`;
